aggregator/entry.go

```diff
@@ -612,8 +616,14 @@ func (e *Entry) addForwarded(
 	}
 
 	// Check if we should update metadata, and add metric if not.
-	if !e.shouldUpdateForwardMetadataWithLock(metadata) {
-		err := e.addForwardedWithLock(metric, metadata.SourceID)
+	key := aggregationKey{
```

Comment:
Maybe make a constructor for new aggregationKey and reuse in all the places? in case you added a field to the struct and forgot to assign value in some places.

Reply:
Talked offline, decided to keep this as is due to its limited use in `entry.go` and `elem_base.go` and convenience to not have to provide all values and can rely on default values for initialization.